burn-import/src/onnx/to_burn.rs

```diff
@@ -287,6 +287,10 @@ impl Tensor {
             TensorData::Int32(val) => DataSerialize::new(val, self.shape.unwrap()).convert(),
             TensorData::Int64(val) => DataSerialize::new(val, self.shape.unwrap()).convert(),
             TensorData::String(_) => panic!("String tensor unsuported"),
+            TensorData::Bool(val) => {
+                DataSerialize::new(val.into_iter().map(u8::from).collect(), self.shape.unwrap())
+                    .convert()
```

Comment:
Because `ToPrimitive` is not implemented for `bool`, `Element` looks like not able to be implemented for `bool`.
What do you think?

Reply:
Yep, `bool` doesn't implement `Element`. It causes problem sometimes, but `Element` really represent numeric types and bool isn't. 

Maybe look here https://stackoverflow.com/questions/72572544/how-to-convert-a-bool-array-into-a-byte-array for help on how to decode `Vec<u8>` into `Vec<bool>`. There are some manipulations to do. Not sure if there is a crate for that, but you could create two functions that convert `Vec<bool>` <=> `Vec<u8>` and test the flow properly. Those functions could be in `burn-tensor` since it can be used anywhere.

Reply:
I have used bytemuck utility (crate) successfully in casting bytes into native types. Here is one example in ONNX: https://github.com/burn-rs/burn/blob/cb4f049eae4772b13d8fdcbb85ca0108e9edc3a3/burn-import/src/onnx/from_onnx.rs#L188

ONNX stores some information in bytes (u8s) to preserve the precision, so it should be straight forward to convert back with right tools.

Reply:
@antimora Yes, I know I can use bytemuck in a way like `TensorData::Bool(cast_slice(&tensor.raw_data[..]).to_vec())` if `TensorData::Bool(Vec<u8>)` is its definition, but now it is defined as `TensorData::Bool(Vec<bool>)` and `cast_slice` from bytemuck looks like it does not support `bool` here (raises compile error). This is what I was doing in previous commit https://github.com/burn-rs/burn/pull/389/commits/f926df282afbc96285cd10b4e132d328bf6e7323.

The problem I have in mind is:

> I realized that by converting bools into u8s for DataSerialize::new, we lose information on whether u8 was just an integer value or originally a bool value.

Aren't there any problem that we lose type information on original data?

Reply:
I don't think `bytemuck` supports `bool`, you have to do it manually. I also think that we should not support `into_data_serialize` for bool tensors, similar to how we don't support it for string tensors.